### ISSUE TITLE
Feature Request: MySQL Update on duplicate key

### ISSUE THREAD

**Issue by Dolphindalt** [diesel-rs/diesel#2489]
MySQL insert on duplicate key support such as the following query:
```sql
INSERT INTO table (id, title, message, likes) VALUES(1, "A", "bruh", 21) ON DUPLICATE KEY UPDATE    
title="A", message="bruh", likes=21
```
I could take a shot at implementing this feature if it should be a feature of diesel.

**Comment by weiznich** [comment]
A well written PR would be welcome.

**Comment by starccy** [comment]
If I am not mistaken, there is already such a feature:
```rust
diesel::insert_into(some_table)
    .values(value)
    .on_conflict(some_table::id)
    .do_update()
    .set(some_value);
```

**Comment by weiznich** [comment]
@starccy That's only available for sqlite and postgres yet. Mysql did not support this at the time of implementing this feature.

**Comment by weiznich** [comment]
Closed by #3177
